content/docs/tutorials/aws/ec2-webserver.md

```diff
@@ -55,66 +55,85 @@ aws:region: (us-east-1)
+        filters: [{
+          name: "name",
+          values: ["amzn-ami-hvm-*"],
+        }],
+        owners: ["137112412989"], // This owner ID is Amazon
+        mostRecent: true,
+    });
+    
     let group = new aws.ec2.SecurityGroup("webserver-secgrp", {
         ingress: [
             { protocol: "tcp", fromPort: 22, toPort: 22, cidrBlocks: ["0.0.0.0/0"] },
         ],
     });
-
+    
     let server = new aws.ec2.Instance("webserver-www", {
         instanceType: size,
         securityGroups: [ group.name ], // reference the security group resource above
-        ami: ami,
+        ami: ami.id,
     });
-
+    
     exports.publicIp = server.publicIp;
     exports.publicHostName = server.publicDns;
     ```
 
     ```typescript
-    const aws = require("@pulumi/aws");
-
+    import * as aws from "@pulumi/aws";
+    
     const size = "t2.micro";     // t2.micro is available in the AWS free tier
-    const ami  = "ami-0ff8a91507f77f867"; // AMI for Amazon Linux in us-east-1 (Virginia)
-
+    
+    const ami = aws.getAmi({
+        filters: [{
+            name: "name",
+            values: ["amzn-ami-hvm-*"],
+        }],
+        owners: ["137112412989"], // This owner ID is Amazon
+        mostRecent: true,
+    });
+    
     const group = new aws.ec2.SecurityGroup("webserver-secgrp", {
         ingress: [
             { protocol: "tcp", fromPort: 22, toPort: 22, cidrBlocks: ["0.0.0.0/0"] },
         ],
     });
-
+    
     const server = new aws.ec2.Instance("webserver-www", {
         instanceType: size,
         securityGroups: [ group.name ], // reference the security group resource above
-        ami: ami,
+        ami: ami.id,
     });
-
-    exports.publicIp = server.publicIp;
-    exports.publicHostName = server.publicDns;
+    
+    export const publicIp = server.publicIp;
+    export const publicHostName = server.publicDns;
     ```
 
     ```python
     import pulumi
-    from pulumi_aws import ec2
-    ami = 'ami-0ff8a91507f77f867' # AMI for Amazon Linux in us-east-1 (Virginia)
-
+    import pulumi_aws as aws
+    
     size = 't2.micro'
-
-    group = ec2.SecurityGroup('webserver-secgrp',
-        description='Enable HTTP access',
-        ingress=[
-            { 'protocol': 'tcp', 'from_port': 22, 'to_port': 22, 'cidr_blocks': ['0.0.0.0/0'] }
-        ])
-
-    server = ec2.Instance('webserver-www',
-        instance_type=size,
-        security_groups=[group.name], # reference security group from above
-        ami=ami)
-
+    
+    ami = aws.get_ami(most_recent="true",
+                      owners=["137112412989"],
+                      filters=[{"name":"name","values":["amzn-ami-hvm-*"]}])
+    
+    group = aws.ec2.SecurityGroup('webserver-secgrp',
+                              description='Enable HTTP access',
+                              ingress=[
+                                  { 'protocol': 'tcp', 'from_port': 22, 'to_port': 22, 'cidr_blocks': ['0.0.0.0/0'] }
+                              ])
```

Comment:
I suspect this formatting will make this not fit horizontally on the screen given our narrow layout.  Worth reformatting to try to ensure the code is readable in the tutorial?

Reply:
Done!